libs/partners/prompty/langchain_prompty/langchain.py

```diff
@@ -1,33 +1,34 @@
-from typing import Any, Dict, Literal
+from typing import Any, Dict
 
 from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder
 from langchain_core.runnables import Runnable, RunnableLambda
 
+from .parsers import RoleMap
 from .utils import load, prepare
 
 
 def create_chat_prompt(
     path: str,
     input_name_agent_scratchpad: str = "agent_scratchpad",
-    template_format: Literal["f-string", "mustache", "jinja2"] = "f-string",
 ) -> Runnable[Dict[str, Any], ChatPromptTemplate]:
     """Create a chat prompt from a Langchain schema."""
 
     def runnable_chat_lambda(inputs: Dict[str, Any]) -> ChatPromptTemplate:
         p = load(path)
         parsed = prepare(p, inputs)
+        # Parsed messages have been templated
+        # Convert to Message objects to avoid templating attempts in ChatPromptTemplate
         lc_messages = []
         for message in parsed:
-            lc_messages.append((message["role"], message["content"]))
+            message_class = RoleMap.get_message_class(message["role"])
+            lc_messages.append(message_class(content=message["content"]))
 
         lc_messages.append(
             MessagesPlaceholder(
                 variable_name=input_name_agent_scratchpad, optional=True
             )  # type: ignore[arg-type]
         )
-        lc_p = ChatPromptTemplate.from_messages(
-            lc_messages, template_format=template_format
```

Comment:
Can you clarify why specifying `template_format` with a default of `"f-string"` is breaking? It looks like that is the default in `ChatPromptTemplate.from_messages`:
https://github.com/langchain-ai/langchain/blob/d6c4803ab068e3fb20fb84ed27f96a6ae33a02d4/libs/core/langchain_core/prompts/chat.py#L1147-L1151

Or is it just unrelated?

Reply:
You're right that's it's unrelated.

Here the issue is that before initializing the `ChatPromptTemplate`, Prompty has already parsed and templated the prompt using the user input.

When invoking `ChatPromptTemplate`, it tries to template the messages and interprets text inside `{ }` (or `{{ }}`) as missing variables. The fix proposed in this PR avoids this templating by converting to `Messages` objects, which are already templated.

Since we skip this second templating entirely, it does not make sense to expose one of its parameters to the user, as it won't have any effect. You can select the template engine used by Prompty directly in the `.prompty`s file YAML header.